lib/spack/spack/test/spec_yaml.py

```diff
@@ -167,6 +162,7 @@ def test_ordered_read_not_required_for_consistent_dag_hash(
     don't want to require them to be serialized in order. This
     ensures that is not required.
     """
+    hash_type = ht.dag_hash
```

Comment:
Since `ht.dag_hash` is the default value for the `hash=` argument for both `Spec.to_json` and `Spec.to_yaml` can we avoid setting `hash_type` and simplify a bit this test?